### PR TITLE
Fix issue with Jetpack connection in site credentials flow

### PR DESCRIPTION
**Pull request by hichamboushaba**
### Description
In the PR #7485, I added ability to auto-authenticate the user to their WordPress.com account when they want to connect their Jetpack account, but I missed the case of when we don't have WordPress.com credentials yet, this caused the flow to break when using site credentials for login.

This PR fixes this by fall backing to load unauthenticated URL when we don't have the data to `POST` to login.

@spencertransier heads up, this PR fixes an issue in the current RC, it's not super urgent though, so if you want to hold a bit after the merge to wait for other fixes, feel free to do it.

### Testing instructions
Follow the testing steps of #7364 and confirm you can execute the flow correctly using this branch.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt [hunk 1]
```diff
@@ -989,7 +989,7 @@ class LoginActivity :
     @Parcelize
     private data class ConnectSiteInfo(
         val isWPCom: Boolean,
-        val isJetpackInstalled: Boolean,
+        val isJetpackConnected: Boolean,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt:992]: The LoginActivity changes are unrelated to the bug, it's just fixing a typo about the name of the variable.

**Review by 0nko — COMMENTED** [review]
After step 6, I'm redirected to a site login screen:



~I'm assuming that's incorrect?~ 
Update: The user role was incorrect.

**Review by 0nko — APPROVED** [review]
:shipit:
